core/domain/classifier_domain_test.py

```diff
@@ -76,6 +76,246 @@ def test_to_dict(self):
             expected_training_job_dict,
             observed_training_job.to_dict())
 
+    def test_update_status_from_failed_to_new_with_successful_update(self):
+        training_job_dict = {
+            'job_id': 'exp_id1.SOME_RANDOM_STRING',
+            'algorithm_id': 'TextClassifier',
+            'interaction_id': 'TextInput',
+            'exp_id': 'exp_id1',
+            'exp_version': 1,
+            'next_scheduled_check_time':
+                datetime.datetime.strptime(
+                    '2017-08-11 12:42:31', '%Y-%m-%d %H:%M:%S'),
+            'state_name': 'a state name',
+            'status': 'FAILED',
+            'training_data': [
+                {
+                    'answer_group_index': 1,
+                    'answers': ['a1', 'a2']
+                },
+                {
+                    'answer_group_index': 2,
+                    'answers': ['a2', 'a3']
+                }
+            ],
+            'classifier_data': {},
+            'data_schema_version': 1
+        }
+        training_job_obj = self._get_training_job_from_dict(
+            training_job_dict)
+        training_job_dict['status'] = 'NEW'
```

Comment:
Why is this needed? Can we not pass `NEW` directly in the next line?

Also instead of using `NEW` directly, use `feconf.TRAINING_JOB_STATUS_NEW`, so that the test does not need update even if we change the constant value later.

Ditto for all cases below.

Reply:
Regarding the first question, yes it can be directly passed but this is how all the other tests were written in the test file when I started working on this issue. So I thought it would be better to adhere to this style as it does help to understand the code in a much better and simpler way. What do you suggest regarding this?

Regarding the second doubt, my bad. I'll make the changes right away

Reply:
> Regarding the first question, yes it can be directly passed but this is how all the other tests were written in the test file when I started working on this issue.

Thanks for pointing this out! Can you please make the changes as I have suggested, everywhere in the test file.

Reply:
Done